### PR TITLE
[client] Add some basic testing and Network

### PR DESCRIPTION
**Pull request by gregnazario**
- Adds basic network configurations for known networks
- Adds a full e2e test of transaction submission and checking
- Merges RestClient and Faucet client like the TS SDK

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -29,6 +68,92 @@ type RestClient struct {
 	baseUrl url.URL
 }
 
+func NewNetworkClient(config NetworkConfig) (client *Client, err error) {
```
Comment (brianolson) [inline client.go:71]: does it make sense to split this into 'new for template' for devnet/testnet/mainnet and 'new for URLs' ?
Maybe 'NewNetworkClient' should be 'NewClient(netname)', and maybe 'NewCustomClient(url,url,url,url)' ?

File: client.go [hunk 2]
```diff
@@ -29,6 +68,92 @@ type RestClient struct {
...
+		faucetUrl,
+	}
+	client = &Client{
+		*restClient,
+		*faucetClient,
+	}
+	return
+}
+
+// TODO: Deprecate
```
Comment (brianolson) [inline client.go:156]: just do it, only cmd/goclient/goclient.go to fix.

File: faucet.go [hunk 3]
```diff
@@ -49,6 +49,40 @@ func truthy(x any) bool {
...
+	}
+	if faucetClient.restClient == nil {
+		slog.Debug("FundAccount no txns to wait for")
+		// no Aptos client to wait on txn completion
+		return nil
+	}
+	slog.Debug("FundAccount wait for txns", "ntxn", len(txnHashes))
+	return faucetClient.restClient.WaitForTransactions(txnHashes)
+}
+
```
Comment (brianolson) [inline faucet.go:85]:
```suggestion
// TODO: deprecate/remove FundAccount(), prefer client struct version
```

File: client.go (hunk 1 above)
Comment (gregnazario) [reply client.go:71]: I'm not really in tune with what is "go like"

I suspect more configuration will have to be added here... like optional timeouts, API keys etc.

This provides the flexibility to add in future versions extra information that allows for that.  Either that or we use a builder pattern, where we allow users to add more things as they go

File: client.go (hunk 2 above)
Comment (gregnazario) [reply client.go:156]: Will do.

File: faucet.go (hunk 3 above)
Comment (gregnazario) [reply faucet.go:85]: Will just get rid of the old faucet, didn't want to conflict with your work

**Comment by gregnazario** [comment]
I'm forcing this one in @brianolson

**Pull request merged**